### ISSUE TITLE
Use vLLM's native Outlines integration for `outlines.serve.serve`

### ISSUE THREAD

**Issue by rlouf** [dottxt-ai/outlines#740]
Now that Outlines was integrated in vLLM we can remove the server code from Outlines and instead import vLLM's code for `outlines.serve.serve`, which should simplify the codebase and ensure that the integration does not break unexpectedly in the future.

**Comment by saattrupdan** [comment]
I guess this only includes the server code, as you mention, and not the `outlines.integrations.vllm` code, right? As the new vLLM `guided_decoding` module that you're referring to only really supports OpenAI server usage, not "vanilla usage" of the `vllm.LLM` class.

**Comment by br3no** [comment]
@saattrupdan the logit processors provided in `outlines.integrations.vllm` can be replaced with those in https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/guided_decoding/outlines_logits_processors.py.

**Comment by saattrupdan** [comment]
> @saattrupdan the logit processors provided in `outlines.integrations.vllm` can be replaced with those in https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/guided_decoding/outlines_logits_processors.py.

Gotcha! So solving this issue amounts to deprecating the vLLM logits processors in Outlines and changing to the ones from vLLM directly, right?

**Comment by br3no** [comment]
I think so, yes.
